src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityProtocol.cs

```diff
@@ -0,0 +1,639 @@
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+using CoreWCF.IdentityModel;
+using CoreWCF.IdentityModel.Policy;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.IdentityModel.Tokens;
+using CoreWCF.Runtime;
+using CoreWCF.Security.Tokens;
+
+namespace CoreWCF.Security
+{
+    // See SecurityProtocolFactory for contracts on subclasses etc
+    abstract class SecurityProtocol : ISecurityCommunicationObject
+    {
+        private static ReadOnlyCollection<SupportingTokenProviderSpecification> s_emptyTokenProviders;
+        private Dictionary<string, Collection<SupportingTokenProviderSpecification>> _mergedSupportingTokenProvidersMap;
+       // private ChannelParameterCollection _channelParameters;
```

Comment:
It looks like this is client only so you can remove all the commented out code that references this